rules/DowngradePhp80/Rector/Class_/DowngradePropertyPromotionRector.php

```diff
@@ -79,17 +79,18 @@ public function getNodeTypes(): array
      */
     public function refactor(Node $node): ?Node
     {
-        $oldComments = $this->getOldComments($node);
-        $promotedParams = $this->resolvePromotedParams($node);
+        $constructorClassMethod = $node->getMethod(MethodName::CONSTRUCT);
+        if (! $constructorClassMethod instanceof ClassMethod) {
+            return [];
+        }
```

Comment:
moved this condition from the inner methods up to the top, as it indicates an early exit condition